cpp/src/arrow/compute/exec/key_hash.cc

```diff
@@ -198,8 +198,10 @@ void Hashing::hash_varlen_helper(uint32_t length, const uint8_t* key, uint32_t*
       last_stripe[1] = last_stripe_base[1];
       last_stripe[1] &= mask;
     }
+    uint32_t lanes[4];
+    memcpy(&lanes, &last_stripe, sizeof(last_stripe));
     for (int j = 0; j < 4; ++j) {
-      uint32_t lane = reinterpret_cast<const uint32_t*>(last_stripe)[j];
```

Comment:
Not related to this PR.
Will this int64* to int32* cast make the hash endianness dependent?

Reply:
This doesn't sound like a concern. The hash is only used internally.